pkgs/build-support/fetchdata/default.nix

```diff
@@ -0,0 +1,36 @@
+{ fetchers }:
+
+# One fetcher to rule them all.
+#
+# The input should only be pure nix data, no functions.
+#
+# Example usage:
+#
+#    fetchurl {
+#      url = "http://example.com/archive.tar.gz";
+#      sha256 = "xxx";
+#    };
+#
+# Becomes:
+#
+#    fetchdata {
+#      type = "fetchurl";
+#      url = "http://example.com/archive.tar.gz";
+#      sha256 = "xxx";
+#    }
+#
+# Which means that the attributes can now be imported
+#
+#    fetchdata (import ./source.nix);
+#
+# That's the whole point of this function.
+#
+{ fetcher ? "fetchurl", ... }@attrs:
```

Comment:
Yeah, not very nice. Clearly the goal is to be able to write in "nixon", i.e. without function calls.